src/hardware/serialport/softmodem.cpp

```diff
@@ -29,6 +29,75 @@
 #include "softmodem.h"
 #include "misc_util.h"
 
+static const char phoneValidChars[] = "01234567890*=,;#+>";
+
+class CPhonebookEntry {
+public:
+	CPhonebookEntry(const char *_phone, const char *_address) {
+		safe_strncpy(phone, _phone, sizeof(phone));
+		safe_strncpy(address, _address, sizeof(phone));
+	}
+
+	const char *IsMatchingPhone(const char *input) {
+		if (strcmp(input, phone) != 0)
+			return NULL;
+
+		return address;
+	}
+
+	static char IsValidPhone(const char *input) {
+		for (const char *ch = input; *ch; ch++) {
+			if (!strchr(phoneValidChars, *ch)) {
+				LOG_MSG("SERIAL: Phone %s contains invalid character %c", input, *ch);
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+private:
+	char phone[64];
+	char address[64];
+};
+
+static std::vector<CPhonebookEntry *> phones;
+
+bool MODEM_ReadPhonebook(const char *filename) {
+	LOG_MSG("SERIAL: Loading phonebook from %s", filename);
+
+	FILE *loadfile = fopen(filename, "rt");
+	if (!loadfile) return false;
+	char linein[512];
+	while (fgets(linein, 512, loadfile)) {
+		char *line = trim(linein);
+		char *phone = StripWord(line);
+		char *address = StripWord(line);
+
+		if (*phone && *address) {
+			// Check phone number for characters ignored by Hayes modems.
+			if (!CPhonebookEntry::IsValidPhone(phone))
+				continue;
+
+			LOG_MSG("SERIAL: Mapped phone %s to address %s", phone, address);
+			CPhonebookEntry *pbEntry = new CPhonebookEntry(phone, address);
+			phones.push_back(pbEntry);
+		}
+	}
+	fclose(loadfile);
+	return true;
+}
+
+static const char *MODEM_CheckPhonebook(const char *input) {
+	for (std::vector<CPhonebookEntry *>::iterator it = phones.begin(); it != phones.end(); it++) {
+		CPhonebookEntry *pbEntry = *it;
+		const char *address = pbEntry->IsMatchingPhone(input);
```

Comment:
```
for (const auto& entry: phones) {
    const char *address = entry.IsMatchingPhone(input);
    ...
```

Reply:
*+1; Just use this formatting, please:*
```
for (const auto &entry : phones) {
	const char *address = entry.IsMatchingPhone(input);
```

Reply:
This is also the loop that's added an issue to the PVS report:

![2020-03-20_15-07](https://user-images.githubusercontent.com/1557255/77209856-aa31ae80-6abc-11ea-8d83-bc8c07768dda.png)